File: LLM-Translations/GPT-4/avatar/Python/C/atcoder_AGC014_D.c
#include <stdio.h>
#include <stdlib.h>

#define MAXN 200005

int N;
int adj[MAXN];
int par[MAXN];
int child_num[MAXN];
int seen[MAXN];
int que[MAXN];
int head = 0, tail = 0;

void push(int x) {
    que[tail++] = x;
}

int pop() {
    return que[head++];
}

int main() {
    scanf("%d", &N);
    for (int i = 0 ; i < N - 1 ; ++i) {
        int a, b;
        scanf("%d%d", &a, &b);
        adj[a].push_back(b);
        adj[b].push_back(a);
    }
    push(1);
    seen[1] = 1;
    while (head != tail) {
        int v = pop();
        for (int i = 0 ; i < adj[v].size() ; ++i) {
            int u = adj[v][i];
            if (seen[u] == 0) {
                seen[u] = 1;
                par[u] = v;
                child_num[v]++;
                push(u);
            }
        }
    }
    int seq[MAXN];
    int seq_sz = 0;
    for (int i = 1 ; i <= N ; ++i) {
        if (child_num[i] == 0) {
            seq[seq_sz++] = i;
        }
    }
    while (seq_sz) {
        int c = seq[--seq_sz];
        seen[c] = 0;
        if (seen[par[c]] == 0) {
            printf("First\n");
            return 0;
        }
        seen[par[c]] = 0;
        child_num[par[par[c]]]--;
        if (child_num[par[par[c]]] == 0) {
            seq[seq_sz++] = par[par[c]];
        }
    }
    printf("Second\n");
    return 0;
}
// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Go/atcoder_AGC014_D.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	N, _ := strconv.Atoi(scanner.Text())
	adj := make([][]int, N+1)
	for i := 0; i < N-1; i++ {
		scanner.Scan()
		edges := strings.Split(scanner.Text(), " ")
		a, _ := strconv.Atoi(edges[0])
		b, _ := strconv.Atoi(edges[1])
		adj[a] = append(adj[a], b)
		adj[b] = append(adj[b], a)
	}

	que := []int{1}
	seen := make([]bool, N+1)
	seen[1] = true
	par := make([]int, N+1)
	child_num := make([]int, N+1)
	for len(que) > 0 {
		v := que[0]
		que = que[1:]
		for _, u := range adj[v] {
			if !seen[u] {
				seen[u] = true
				par[u] = v
				child_num[v]++
				que = append(que, u)
			}
		}
	}

	seq := []int{}
	for i := 1; i < N+1; i++ {
		if child_num[i] == 0 {
			seq = append(seq, i)
		}
	}

	for len(seq) > 0 {
		c := seq[len(seq)-1]
		seq = seq[:len(seq)-1]
		seen[c] = false
		if !seen[par[c]] {
			fmt.Println("First")
			return
		}
		seen[par[c]] = false
		child_num[par[par[c]]]--
		if child_num[par[par[c]]] == 0 {
			seq = append(seq, par[par[c]])
		}
	}
	fmt.Println("Second")
}
// End of Code


Please note that input/output handling in Go is different than Python, so it may differ from your original code. Also, Go doesn't have deque in its standard library, so I've used slices instead. The pop operation from the end of slice is O(1) similar to deque in python.
